crates/storage/errors/src/db.rs

```diff
@@ -97,18 +98,40 @@ pub struct DatabaseWriteError {
     pub key: Vec<u8>,
 }
 
+impl fmt::Display for DatabaseWriteError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+           f, 
+           "write operation {} failed for key \"{}\" in table {}: {}", 
+           self.operation, 
+           reth_primitives::hex::encode(&self.key), 
+           self.table_name, 
+           self.info
+        )
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for DatabaseWriteError {}
+
 /// Database write operation type.
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, derive_more::Display)]
+#[display(fmt = "DatabaseWriteOperation: {}")]
 pub enum DatabaseWriteOperation {
     /// Append cursor.
+    #[display(fmt = "CursorAppend")]
     CursorAppend,
     /// Upsert cursor.
+    #[display(fmt = "CursorUpsert")]
     CursorUpsert,
     /// Insert cursor.
+    #[display(fmt = "CursorInsert")]
     CursorInsert,
     /// Append duplicate cursor.
+    #[display(fmt = "CursorAppendDup")]
     CursorAppendDup,
     /// Put.
+    #[display(fmt = "Put")]
     Put,
 }
```

Comment:
there should be no need for it, you can use the `Debug` formatting via `{:?}` in `Display for DatabaseWriteError` implementation

Reply:
you were right, done